elyra/pipeline/processor_airflow.py

```diff
@@ -45,6 +49,29 @@ class AirflowPipelineProcessor(RuntimePipelineProcess):
     def type(self):
         return self._type
 
+    @property
+    def component_registry(self) -> str:
+        return self._component_registry_location
+
+    @property
+    def component_parser(self) -> ComponentParser:
+        return self._component_parser
+
+    def __init__(self, root_dir, **kwargs):
+        super().__init__(root_dir, **kwargs)
+        # then sys.prefix, where installed files will reside (factory data)
+        self._component_registry_location =  \
+            os.path.join(jupyter_core.paths.ENV_JUPYTER_PATH[0],
+                         'components',
+                         f"{self._type}_component_catalog.json")
+
+        if not os.path.exists(self._component_registry_location):
+            raise FileNotFoundError(f'Invalid component_id catalog path {self._component_registry_location}'
+                                    f' for {self._type} processor')
```

Comment:
Rather than use a local variable to house the location, the location composition and its existence check could be done from the superclass constructor since `self._type` will have the appropriate value.  The same could be done for the parser provided the instance is created prior to `super.__init__()`, then you could have the creation of the `ComponentRegistry` instance done in the superclass constructor.

Regardless, the two "component" properties (if they will remain) should be moved to the superclass anyway since kfp does the exact same thing, relatively speaking.